storage/raft_transport.go

```diff
@@ -386,3 +410,23 @@ func (t *RaftTransport) SendAsync(req *RaftMessageRequest) bool {
 		return false
 	}
 }
+
+// SendSync sends a raft message and waits for an acknowledgement.
+func (t *RaftTransport) SendSync(ctx context.Context, req *RaftMessageRequest) error {
```

Comment:
I'll have to refactor a bit in order to merge this with my pending transport change. Is there any reason we shouldn't use the same breaker we use for the async send?